internal/k8stest/k8s_objects.go

```diff
@@ -36,5 +36,12 @@ func DeleteObject(client *dynamic.DynamicClient, obj *unstructured.Unstructured)
 		Version:  gvk.Version,
 		Resource: strings.ToLower(gvk.Kind + "s"),
 	}
-	return client.Resource(gvr).Namespace(obj.GetNamespace()).Delete(context.Background(), obj.GetName(), metav1.DeleteOptions{})
+
+	options := metav1.DeleteOptions{}
+	policy := metav1.DeletePropagationBackground
+	if gvk.Kind == "Job" {
```

Comment:
Noticed pods left orphaned from the k8sattributeprocessor e2e test which uses jobs for generating telemetry. This makes sure the pods are deleted since we have other e2e tests running in the same namespace after the k8sattributeprocessor test.

Reply:
We should properly clean up the test, but is it also possible to run the tests in different namespaces?

Reply:
Yeah, can be done. I think better to have a different issue for this since it would need updates to test manifests and data too.